packages/mdc-menu/simple/foundation.js

```diff
@@ -150,109 +173,27 @@ class MDCSimpleMenuFoundation extends MDCFoundation {
   }
 
   /**
-   * Calculates transition delays for individual menu items, so that they fade in one at a time.
-   * @private
+   * @param {!Corner} corner Default anchor corner alignment of top-left menu corner.
    */
-  applyTransitionDelays_() {
-    const {BOTTOM_LEFT, BOTTOM_RIGHT} = MDCSimpleMenuFoundation.cssClasses;
-    const numItems = this.adapter_.getNumberOfItems();
-    const {height} = this.dimensions_;
-    const transitionDuration = MDCSimpleMenuFoundation.numbers.TRANSITION_DURATION_MS / 1000;
-    const start = MDCSimpleMenuFoundation.numbers.TRANSITION_SCALE_ADJUSTMENT_Y;
-
-    for (let index = 0; index < numItems; index++) {
-      const {top: itemTop, height: itemHeight} = this.adapter_.getYParamsForItemAtIndex(index);
-      this.itemHeight_ = itemHeight;
-      let itemDelayFraction = itemTop / height;
-      if (this.adapter_.hasClass(BOTTOM_LEFT) || this.adapter_.hasClass(BOTTOM_RIGHT)) {
-        itemDelayFraction = ((height - itemTop - itemHeight) / height);
-      }
-      const itemDelay = (start + itemDelayFraction * (1 - start)) * transitionDuration;
-      // Use toFixed() here to normalize CSS unit precision across browsers
-      this.adapter_.setTransitionDelayForItemAtIndex(index, `${itemDelay.toFixed(3)}s`);
-    }
+  setAnchorCorner(corner) {
+    this.anchorCorner_ = corner;
   }
 
   /**
-   * Removes transition delays from menu items.
-   * @private
+   * @param {!AnchorMargin} margin 4-plet of margins from anchor.
    */
-  removeTransitionDelays_() {
-    const numItems = this.adapter_.getNumberOfItems();
-    for (let i = 0; i < numItems; i++) {
-      this.adapter_.setTransitionDelayForItemAtIndex(i, null);
+  setAnchorMargin(margin) {
```

Comment:
I wanted to allow people to call setAnchorMargin({top: 5, bottom: -5}) without having to worry about right/left. This API kind of follows the css style specification in that it does not reset margin to 0 for sides you have not specified.

Reply:
I'd argue that in your example, people _do_ have to worry about whether right/left have been set previously. i.e. previous calls to this `set` API can have side effects on future calls, and if they want to be sure that _only_ the margins they pass are in effect, they have to pass all 4 sides and zero out the ones they don't want, which seems less convenient in the long run.

IMO it's easier to consume `set` APIs that function as resets, not potentially additive across subsequent calls.

Reply:
I do not expect people to change margins more than once, frankly. It has to do with anchor geometry and should probably not change between invocations. So, being able to specify just the margins that one cares about is useful. One can always set all 4 to be sure. If you feel strongly about this, I can add assert for all 4 fields to be there on each set.
I do not think that not including a side should reset that side to 0 margin.

Reply:
Discussed offline, but in summary: I definitely agree that this will be called only once in most cases, which makes me feel a little guilty about harping on it :) but resetting the unspecified states seems cleaner from a setter API perspective, and also can be accomplished with less code, which seems like a win-win to me.

Reply:
Single liner breaks closure test, since margin is of type AnchorMargin and not Object.